identity-access-manager-keycloak/swarm.sh

```diff
@@ -16,46 +16,81 @@ readonly ROOT_PATH
 . "${ROOT_PATH}/utils/docker-utils.sh"
 . "${ROOT_PATH}/utils/log.sh"
 
+await_postgres_start() {
+  log info "Waiting for Postgres to start up before KeyCloak"
+
+  docker::await_container_startup keycloak-postgres-1
+  docker::await_container_status keycloak-postgres-1 Running
+
+  if [[ "$STATEFUL_NODES" == "cluster" ]]; then
+    docker::await_container_startup keycloak-postgres-2
+    docker::await_container_status keycloak-postgres-2 Running
+
+    docker::await_container_startup keycloak-postgres-3
+    docker::await_container_status keycloak-postgres-3 Running
+  fi
+}
+
 main() {
+  if [ "${STATEFUL_NODES}" == "cluster" ]; then
+    log info "Running Identity Access Manager Keycloak package in Cluster node mode"
+    postgres_cluster_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose-postgres.cluster.yml"
+  else
+    log info "Running Identity Access Manager Keycloak package in Single node mode"
+    postgres_cluster_compose_param=""
+  fi
+
   if [[ "${MODE}" == "dev" ]]; then
     log info "Running Identity Access Manager Keycloak package in DEV mode"
     keycloak_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.dev.yml"
+    postgres_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose-postgres.dev.yml"
   else
     log info "Running Identity Access Manager Keycloak package in PROD mode"
     keycloak_dev_compose_param=""
+    postgres_dev_compose_param=""
   fi
 
   if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
+    log info "Setting config digests"
+    config::set_config_digests "$COMPOSE_FILE_PATH"/docker-compose.yml
+
+    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose-postgres.yml $postgres_cluster_compose_param $postgres_dev_compose_param instant" "Failed to deploy KeyCloak Postgres"
```

Comment:
I think this should be updated to use the new function:
`docker::deploy_service`

Reply:
I think the PR containing that function is not yet merged. 

@nour-borgi should revisit this package to apply the latest update to the script.

Reply:
Yes will do no worries about this :)